### PR TITLE
New lookup-plugin: `dig' for performing DNS lookups

### PR DESCRIPTION
**Pull request by jpmens**
This makes `dnstxt.py` obsolete, and I recommend it be removed.

### CODE REVIEW

File: lib/ansible/runner/lookup_plugins/dig.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        except dns.resolver.NoAnswer:
+            responses.append("")
+        except dns.resolver.Timeout:
+            responses.append('')
+        except dns.exception.DNSException, e:
+            raise errors.AnsibleError("dns.resolver unhandled exception", e)
+
+        for r in responses:
+            ret.append(r)
+        return ret
```
Comment (bcoca) [inline lib/ansible/runner/lookup_plugins/dig.py]: I'm not sure ret/responses are in the correct scope, it looks like you wanted responses inside a for loop (terms one?) in which case indentation is wrong.

**Comment by bcoca** [comment]
I'm tempted to rename it to just 'dns' also to make the record type a named parameter qtype='A' (default)

File: lib/ansible/runner/lookup_plugins/dig.py (hunk 1 above)
Comment (jpmens) [inline lib/ansible/runner/lookup_plugins/dig.py]: Tested. Works fine here.

**Comment by jpmens** [comment]
You can't call it 'dns' -- that clashes with python 'dns' module requirement.

'A' is default as it is.

File: lib/ansible/runner/lookup_plugins/dig.py (hunk 1 above)
Comment (jpmens) [inline lib/ansible/runner/lookup_plugins/dig.py]: I've double-checked @bcoca -- this is definitely ok as it is.

**Comment by bcoca** [comment]
duh! of course you cannot call it dns .... (what idiot suggested that?)

as for the qtype, yes, A is already the default, but what I wanted was it being passed as a named parameter instead of splitting the string as it allows for more flexible calling (w/o having to manipulate domain/ip data).

**Comment by jpmens** [comment]
I've added a commit which adds `qtype=xxx` support. Qtype can be lower or uppercase. Old behaviour remains though, so these are all valid:

```
example.com                            A record
example.com/A
example.com qtype=a

example.net/TXT                        TXT record
example.net  qtype=txt
```

**Comment by bcoca** [comment]
perfect!

**Comment by jpmens** [comment]
@bcoca How's this in addition?

``` jinja
{% set data = lookup('dig', 'iis.se. qtype=a', wantlist=True)[0] -%}
{{ data }}

{% set data = lookup('dig', 'iis.se. qtype=a flat=0', wantlist=True)[0] -%}
Domain...: {{ data.owner }}
Type.....: {{ data.type }}
Address..: {{ data.address }}
TTL......: {{ data.ttl }}

{% set data = lookup('dig', 'iis.se. qtype=mx flat=0', wantlist=True) -%}
{{ data | pprint }}
```

produces

```
91.226.36.46

Domain...: iis.se.
Type.....: A
Address..: 91.226.36.46
TTL......: 60

[{'exchange': 'mx2.iis.se.',
  'owner': 'iis.se.',
  'prio': 5,
  'ttl': 60,
  'type': 'MX'},
 {'exchange': 'mx1.iis.se.',
  'owner': 'iis.se.',
  'prio': 5,
  'ttl': 60,
  'type': 'MX'}]
```

Setting `flat=0` on a per/query basis returns the data as a dict.

**Comment by bcoca** [comment]
Very nice!

**Comment by clickfreak** [comment]
:+1:

**Comment by jpmens** [comment]
OK, as far as I'm concerned this is now feature complete. :-)

**Comment by jpmens** [comment]
Thank you, @bcoca !

**Pull request merged**